### PR TITLE
Add com.github.qarmin.szyszka

### PR DESCRIPTION
**Pull request by qarmin**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by qarmin** [comment]
bot, build com.github.qarmin.szyszka

File: com.github.qarmin.szyszka.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  - cargo --offline build --release
+  - install -Dm755 ./target/release/szyszka -t /app/bin/
+  - install -Dm644 ./data/icons/com.github.qarmin.szyszka.svg -t /app/share/icons/hicolor/scalable/apps/
+  - install -Dm644 ./data/com.github.qarmin.szyszka.desktop -t /app/share/applications/
+  - install -Dm644 ./data/com.github.qarmin.szyszka.metainfo.xml -t /app/share/metainfo
+  sources:
+  - cargo-sources.json
+  - type: git
+    url: https://github.com/qarmin/szyszka.git
+    tag: 2.0.0
```
Comment (hfiguiere) [inline com.github.qarmin.szyszka.yaml:32]: There should *also* be the commit sha for the tag. This protect against retags.

File: com.github.qarmin.szyszka.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+runtime-version: '40'
+sdk: org.gnome.Sdk
+sdk-extensions:
+- org.freedesktop.Sdk.Extension.rust-stable
+command: szyszka
+finish-args:
+- "--share=ipc"
+- "--socket=fallback-x11"
+- "--socket=wayland"
+- "--filesystem=host"
```
Comment (hfiguiere) [inline com.github.qarmin.szyszka.yaml:12]: why do you need host permissions here?

File: com.github.qarmin.szyszka.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+- "--device=dri"
+build-options:
+  append-path: "/usr/lib/sdk/rust-stable/bin"
+  env:
+    CARGO_HOME: "/run/build/szyszka_gui/cargo"
```
Comment (hfiguiere) [inline com.github.qarmin.szyszka.yaml:17]:
```suggestion
    CARGO_HOME: "/run/build/szyszka/cargo"
```

This likely explain the failure.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.qarmin.szyszka.yaml (hunk 2 above)
Comment (barthalion) [reply com.github.qarmin.szyszka.yaml:12]: It's a file renamer so that's likely why.

Comment (hfiguiere) [reply com.github.qarmin.szyszka.yaml:12]: so `home`, `/media` and `/run/media`, right?

Comment (barthalion) [reply com.github.qarmin.szyszka.yaml:12]: That's also fine, just not sure if it really matters for an offline app. I don't have a strong opinion here.

**Comment by qarmin** [comment]
bot, build com.github.qarmin.szyszka

File: com.github.qarmin.szyszka.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  - install -Dm755 ./target/release/szyszka -t /app/bin/
+  - install -Dm644 ./data/icons/com.github.qarmin.szyszka.svg -t /app/share/icons/hicolor/scalable/apps/
+  - install -Dm644 ./data/com.github.qarmin.szyszka.desktop -t /app/share/applications/
+  - install -Dm644 ./data/com.github.qarmin.szyszka.metainfo.xml -t /app/share/metainfo
+  sources:
+  - cargo-sources.json
+  - type: git
+    url: https://github.com/qarmin/szyszka.git
+    tag: 2.0.0
+    sha: 827b94496f09f0d0bea14e0d2f44bdf0496b1f76
```
Comment (hfiguiere) [inline com.github.qarmin.szyszka.yaml:36]: It's commit" `(flatpak-builder:129186): flatpak-builder-WARNING **: 13:59:35.858: Unknown property sha for type BuilderSourceGit`

```suggestion
    commit: 827b94496f09f0d0bea14e0d2f44bdf0496b1f76
```

**Comment by qarmin** [comment]
bot, build com.github.qarmin.szyszka

**Comment by teohhanhui** [comment]
The repository created is (wrongly) a clone of the `flathub/flathub` repository.

So it contains all the files that shouldn't be there, e.g.

* `.github`
* `CONTRIBUTING.md`
* `COPYING`
* `README.md`

And as such it also contains the wrong commit history...

**Comment by teohhanhui** [comment]
Looks like it's because the `new-pr` branch was not used.

Perhaps flathubbot should have a check for that?

**Pull request closed without merging**